File: src/main/java/shape/Rectangle.java
package shape;

public class Rectangle {
    private Point pointOne;
    private Point pointTwo;

    public Rectangle(Point pointOne, Point pointTwo) {
        this.pointOne = pointOne;
        this.pointTwo = pointTwo;

        // Check if the area is 0, if so throw an exception
        double area = Math.abs(pointTwo.GetX() - pointOne.GetX()) * Math.abs(pointTwo.GetY() - pointOne.GetY());
        if (area <= 0) {
            throw new IllegalArgumentException("The two points cannot be the same.");
        }
    }

    public Point GetPointOne() {
        return pointOne;
    }
    public Point GetPointTwo() {
        return pointTwo;
    }
    public double GetArea() {
        double area = Math.abs(pointTwo.GetX() - pointOne.GetX()) * Math.abs(pointTwo.GetY() - pointOne.GetY());
        return area;
    }

    public void MoveRectangle(double addXvalue, double addYvalue) {
        pointOne.MovePoint(addXvalue, addYvalue);
        pointTwo.MovePoint(addXvalue, addYvalue);
    }
}

File: src/main/kotlin/shape/Rectangle.kt
package shape

import kotlin.math.abs

class Rectangle(private val pointOne: Point, private val pointTwo: Point) {

    init {
        if (GetArea() <= 0) {
            throw IllegalArgumentException("The two points cannot be the same.")
        }
    }

    fun GetPointOne(): Point {
        return pointOne
    }

    fun GetPointTwo(): Point {
        return pointTwo
    }

    fun GetArea(): Double {
        return abs(pointTwo.GetX() - pointOne.GetX()) *
                abs(pointTwo.GetY() - pointOne.GetY())
    }

    fun MoveRectangle(addXvalue: Double, addYvalue: Double) {
        pointOne.MovePoint(addXvalue, addYvalue)
        pointTwo.MovePoint(addXvalue, addYvalue)
    }
}